common/numerated/src/tree.rs

```diff
@@ -236,16 +234,16 @@ impl<T: Numerated> IntervalsTree<T> {
         let end = right_end.max(end);
 
         let Some((left_start, left_end)) = left_interval else {
+            debug_assert!(start <= end, "Must be cause of method it was found");
```

Comment:
if we won't place panics here , than this check seems excess

Reply:
It looks like this check should not be here at all

Reply:
removed and use put